packages/lodestar/src/chain/blocks/processor.ts

```diff
@@ -58,7 +58,10 @@ export class BlockProcessor implements IService {
 
   public async start(): Promise<void> {
     const abortSignal = this.controller.signal;
-    pipe(
+    // An error in this pipe should panic the process, so it's not handled
+    // void indicates that a function is not awaited on purpose
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
```

Comment:
you don't need this eslint ignore. Also error in pipe should't panic process :slightly_smiling_face:

Reply:
I've tried without `eslint-disable-next-line @typescript-eslint/no-floating-promises` and the rule still kicks in, seems it doesn't accept `void` as a valid promise handler

Reply:
Try to explicitly set in eslint rule config:
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-floating-promises.md#options

Reply:
@wemeetagain I thought it was said that pipe had to indeed let the error bubble up.

Reply:
I'm not sure that its accurate to say it panics the process? It'll just be an unhandled promise rejection, right?

In any case, if this pipe errors out, its not great. We stop processing any new blocks.

We might just add a TODO / issue to have more robust error handling here. Either with better error handling inside each segment of the pipeline, or surrounding the pipeline.
The current state of things is: the pipeline doesn't error in practice because there's input validation pre-pipeline. So this is more a hardening / stylistic issue than an urgent practical issue.